File: EveHQ.ItemBrowser/PlugInData.vb
' ========================================================================
' EveHQ - An Eve-Online™ character assistance application
' Copyright © 2005-2012  EveHQ Development Team
' 
' This file is part of EveHQ.
'
' EveHQ is free software: you can redistribute it and/or modify
' it under the terms of the GNU General Public License as published by
' the Free Software Foundation, either version 3 of the License, or
' (at your option) any later version.
'
' EveHQ is distributed in the hope that it will be useful,
' but WITHOUT ANY WARRANTY; without even the implied warranty of
' MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
' GNU General Public License for more details.
'
' You should have received a copy of the GNU General Public License
' along with EveHQ.  If not, see <http://www.gnu.org/licenses/>.
'=========================================================================

Public Class PlugInData
    Implements EveHQ.Core.IEveHQPlugIn

    Public Shared PlugInDataObject As Object
    Public Shared AttributeList As New SortedList
    Public Shared EffectList As New SortedList
    Public Shared Event PluginDataReceived()

    Public Function GetPlugInData(ByVal data As Object, ByVal dataType As Integer) As Object Implements Core.IEveHQPlugIn.GetPlugInData
        Try
            Select Case DataType
                Case 0 ' Item ID
                    PlugInDataObject = data
                    RaiseEvent PluginDataReceived()
            End Select
            Return True
        Catch e As Exception
            Return False
        End Try
    End Function

    Public Function EveHQStartUp() As Boolean Implements Core.IEveHQPlugIn.EveHQStartUp
        Try
            ' Load attribute names
            AttributeList.Clear()
            Dim eveData As DataSet = EveHQ.Core.DataFunctions.GetData("SELECT DISTINCT dgmTypeAttributes.attributeID, dgmAttributeTypes.attributeName FROM dgmAttributeTypes INNER JOIN dgmTypeAttributes ON dgmAttributeTypes.attributeID = dgmTypeAttributes.attributeID;")
            For item As Integer = 0 To eveData.Tables(0).Rows.Count - 1
                If PlugInData.AttributeList.ContainsKey(eveData.Tables(0).Rows(item).Item("attributeName").ToString.Trim) = False Then
                    PlugInData.AttributeList.Add(eveData.Tables(0).Rows(item).Item("attributeName").ToString.Trim, eveData.Tables(0).Rows(item).Item("attributeID").ToString.Trim)
                End If
            Next
            ' Load Effect Names
            EffectList.Clear()
            eveData = EveHQ.Core.DataFunctions.GetData("SELECT DISTINCT dgmTypeEffects.effectID, dgmEffects.effectName FROM dgmEffects INNER JOIN dgmTypeEffects ON dgmEffects.effectID = dgmTypeEffects.effectID;")
            For item As Integer = 0 To eveData.Tables(0).Rows.Count - 1
                If PlugInData.EffectList.ContainsKey(eveData.Tables(0).Rows(item).Item("effectName").ToString.Trim) = False Then
                    PlugInData.EffectList.Add(eveData.Tables(0).Rows(item).Item("effectName").ToString.Trim, eveData.Tables(0).Rows(item).Item("effectID").ToString.Trim)
                End If
            Next
            Return True
        Catch ex As Exception
            Windows.Forms.MessageBox.Show(ex.Message)
            Return False
        End Try
    End Function
    Public Function GetEveHQPlugInInfo() As Core.EveHQPlugIn Implements Core.IEveHQPlugIn.GetEveHQPlugInInfo
        ' Returns data to EveHQ to identify it as a plugin
        Dim EveHQPlugIn As New EveHQ.Core.EveHQPlugIn
        EveHQPlugIn.Name = "EveHQ Item Browser"
        EveHQPlugIn.Description = "Shows Data on all Eve ingame items"
        EveHQPlugIn.Author = "EveHQ Team"
        EveHQPlugIn.MainMenuText = "Item Browser"
        EveHQPlugIn.RunAtStartup = True
        EveHQPlugIn.RunInIGB = False
        EveHQPlugIn.MenuImage = My.Resources.plugin_icon
        EveHQPlugIn.Version = My.Application.Info.Version.ToString
        Return EveHQPlugIn
    End Function

    Public Function IGBService(ByVal igbContext As Net.HttpListenerContext) As String Implements Core.IEveHQPlugIn.IGBService
        Return ""
    End Function

    Public Function RunEveHQPlugIn() As System.Windows.Forms.Form Implements Core.IEveHQPlugIn.RunEveHQPlugIn
        Return New frmItemBrowser
    End Function

    Public Function SaveAll() As Boolean Implements Core.IEveHQPlugIn.SaveAll
        ' No data or settings to save
        Return False
    End Function

End Class

File: EveHQ.PlanetaryInteraction/PlugInData.cs
using EveHQ.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace EveHQ.PlanetaryInteraction
{
    public class PlugInData : IEveHQPlugIn
    {
        #region "Plug-in Interface Properties and Functions"

        EveHQPlugIn IEveHQPlugIn.GetEveHQPlugInInfo()
        {
            EveHQPlugIn eveHQPlugIn = new EveHQPlugIn();
            eveHQPlugIn.Name = "EveHQ Planetary Colonies";
            eveHQPlugIn.Description = "Monitor planetary colonies";
            eveHQPlugIn.Author = "EveHQ Team";
            eveHQPlugIn.MainMenuText = "EveHQ Planetary Colonies";
            eveHQPlugIn.RunAtStartup = true;
            eveHQPlugIn.RunInIGB = true;
            eveHQPlugIn.MenuImage = Properties.Resources.NeoComPlanets;
            eveHQPlugIn.Version = Application.ProductVersion.ToString();
            return eveHQPlugIn;
        }

        Boolean IEveHQPlugIn.EveHQStartUp()
        {
            return true;
        }

        String IEveHQPlugIn.IGBService(HttpListenerContext igbContext)
        {
            return "";
        }

        Form IEveHQPlugIn.RunEveHQPlugIn()
        {
            return new FrmPI();
        }

        Object IEveHQPlugIn.GetPlugInData(Object data, int dataType)
        {
            return null;
        }

        Boolean IEveHQPlugIn.SaveAll()
        {
            return false;
        }

        #endregion

    }
}
